src/golang/lib/engine/utils.go

```diff
@@ -50,3 +59,103 @@ func opFailureError(failureType shared.FailureType, op operator.Operator) error
 func shouldStopExecution(execState *shared.ExecutionState) bool {
 	return execState.Status == shared.FailedExecutionStatus && *execState.FailureType != shared.UserNonFatalFailure
 }
+
+func generateJobManagerConfig(
+	ctx context.Context,
+	dbWorkflowDag *workflow_dag.DBWorkflowDag,
+	aqPath string,
+	vault vault.Vault,
+) (job.Config, error) {
+	switch dbWorkflowDag.EngineConfig.Type {
+	case shared.AqueductEngineType:
+		return &job.ProcessConfig{
+			BinaryDir:          path.Join(aqPath, job.BinaryDir),
+			OperatorStorageDir: path.Join(aqPath, job.OperatorStorageDir),
+		}, nil
+	case shared.K8sEngineType:
+		if dbWorkflowDag.StorageConfig.Type != shared.S3StorageType {
+			return nil, errors.New("Must use S3 storage config for K8s engine.")
+		}
+		awsAccessKeyId, awsSecretAccessKey, err := extractAwsCredentials(dbWorkflowDag.StorageConfig.S3Config)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to extract AWS credentials from file.")
+		}
+
+		k8sIntegrationId := dbWorkflowDag.EngineConfig.K8sConfig.IntegrationId
+		config, err := auth.ReadConfigFromSecret(ctx, k8sIntegrationId, vault)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to read config from vault.")
+		}
+		k8sConfig, err := ParseK8sConfig(config)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to get integration.")
+		}
+		return &job.K8sJobManagerConfig{
+			KubeconfigPath:                   k8sConfig.KubeconfigPath,
+			ClusterName:                      k8sConfig.ClusterName,
+			AwsAccessKeyId:                   awsAccessKeyId,
+			AwsSecretAccessKey:               awsSecretAccessKey,
+			AwsRegion:                        DefaultAwsRegion,
+			FunctionDockerImage:              DefaultFunctionDockerImage,
```

Comment:
Can these images just be constants instead of needing to set them as config fields